File: LEETCODE/LEETCODE-JAVA/1446.java
// 1446. Consecutive Characters
// https://leetcode.com/problems/consecutive-characters

//Method-1 counting
class Solution {
    public int maxPower(String s) {
        int n=s.length(), count=0, ans=0;

        char st[]=s.toCharArray();

        for(int i=1;i<n;i++){
            if(st[i-1]==st[i]){
                count++;
            }
            else{
                count=0;
            }

            ans=Math.max(ans,count);
        }

        return ans+1;
    }
}

File: LEETCODE/LEETCODE-CPP/1446.cpp
// 1446. Consecutive Characters
// https://leetcode.com/problems/consecutive-characters

// Method-1 counting
class Solution
{
public:
    int maxPower(string s)
    {
        int n = s.length(), count = 0, max1 = 0;

        if (n == 1)
            return 1;

        for (int i = 0; i < n - 1; i++)
        {
            if (s[i] == s[i + 1])
            {
                count++;
                max1 = max(max1, count);
            }
            else
                count = 0;
        }
        return max1 + 1;
    }
};